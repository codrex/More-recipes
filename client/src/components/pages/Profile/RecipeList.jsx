import React from 'react';
import PropTypes from 'prop-types';
import List from '../../common/List';
import ListItem from '../../common/ListItem';
import Icon from '../../common/Icon';


/**
 * RecipeList component
 * @param {object} props
 * @return {React} react element
 */
const RecipeList = (props) => {
  const recipes = props.recipes || [];
  const {
    onDeleteIconClicked,
    onEditIconCliked,
    handleClick
  } = props;
  return (
    <List
      className="recipe-list"
    >
      {recipes.length > 0 &&
        <h4 className="display-4 bold header">
          my recipes
        </h4>
      }
      {
        recipes.map((item, i) =>
          (
            <ListItem
              key={item.id}
              index={i}
              content={item.name}
              className="recipe-list-item"
              handleClick={() => handleClick(item)}
            >
              <div>
                <Icon
                  iconClass="fa fa-pencil edit-icon"
                  pointer
                  handleClick={() => onEditIconCliked(recipes[i].id)}
                />
                <Icon
                  iconClass="fa fa-trash-o delete-icon"
                  dataToggle="modal"
                  dataTarget="#modal"
                  pointer
                  handleClick={() => onDeleteIconClicked('Delete recipe', item.id, i)}
                />
              </div>
            </ListItem>
          )
        )}
    </List>
  );
};

RecipeList.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.shape).isRequired,
  onEditIconCliked: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired,
  onDeleteIconClicked: PropTypes.func.isRequired,
};

export default RecipeList;
