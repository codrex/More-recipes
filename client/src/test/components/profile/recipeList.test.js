import React from 'react';
import toJson from 'enzyme-to-json';
import RecipeList from '../../../components/pages/Profile/RecipeList';

const props = {
  recipes: [{
    name: 'recipe',
    category: '',
    views: 1,
    upVotes: 1,
    downVotes: 1,
    id: 1
  }],
  onEditIconCliked: jest.fn(),
  handleClick: jest.fn(),
  onDeleteIconClicked: jest.fn(),
  onFavIconClicked: jest.fn(),
  type: 'createdRecipes',
  index: '1'
};

describe('Recipes list component ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<RecipeList {...props} />);
  });
  test('to match empty snapshot ', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
  test('render as expected when delete and edit icon are clicked ', () => {
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    wrapper.find('ListItem').find('Icon').at(0)
      .simulate('click');
    wrapper.find('ListItem').find('Icon').at(1)
      .simulate('click');
    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});

