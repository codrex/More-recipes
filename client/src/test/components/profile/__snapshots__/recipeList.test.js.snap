// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes list component  render as expected when delete and edit icon are clicked  1`] = `
<RecipeList
  handleClick={[Function]}
  index="1"
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {
        "category": "",
        "downVotes": 1,
        "id": 1,
        "name": "recipe",
        "upVotes": 1,
        "views": 1,
      },
    ]
  }
  type="createdRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        content="recipe"
        handleClick={[Function]}
        index={0}
        key="1"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div
            className="list-item-text lead"
            data-index={0}
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            recipe
          </div>
          <div>
            <Icon
              active=""
              className=""
              dataTarget=""
              dataToggle=""
              handleClick={[Function]}
              iconClass="fa fa-pencil edit-icon"
              onMouseEnter={[Function]}
              parentClass=""
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-pencil edit-icon"
                  data-target=""
                  data-toggle=""
                >
                  <span />
                </i>
              </div>
            </Icon>
            <Icon
              active=""
              className=""
              dataTarget="#modal"
              dataToggle="modal"
              handleClick={[Function]}
              iconClass="fa fa-trash-o delete-icon"
              onMouseEnter={[Function]}
              parentClass=""
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-trash-o delete-icon"
                  data-target="#modal"
                  data-toggle="modal"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</RecipeList>
`;

exports[`Recipes list component  render as expected when delete and edit icon are clicked  2`] = `
<RecipeList
  handleClick={[Function]}
  index="1"
  onDeleteIconClicked={[Function]}
  onEditIconCliked={[Function]}
  onFavIconClicked={[Function]}
  recipes={
    Array [
      Object {
        "category": "",
        "downVotes": 1,
        "id": 1,
        "name": "recipe",
        "upVotes": 1,
        "views": 1,
      },
    ]
  }
  type="createdRecipes"
>
  <List
    className="recipe-list"
    style={Object {}}
  >
    <div
      className="col-12 recipe-list"
      style={Object {}}
    >
      <h4
        className="display-4 bold header"
      >
        my recipes
      </h4>
      <ListItem
        className="recipe-list-item"
        content="recipe"
        handleClick={[Function]}
        index={0}
        key="1"
      >
        <div
          className="lead list-item recipe-list-item"
        >
          <div
            className="list-item-text lead"
            data-index={0}
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            recipe
          </div>
          <div>
            <Icon
              active=""
              className=""
              dataTarget=""
              dataToggle=""
              handleClick={[Function]}
              iconClass="fa fa-pencil edit-icon"
              onMouseEnter={[Function]}
              parentClass=""
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-pencil edit-icon"
                  data-target=""
                  data-toggle=""
                >
                  <span />
                </i>
              </div>
            </Icon>
            <Icon
              active=""
              className=""
              dataTarget="#modal"
              dataToggle="modal"
              handleClick={[Function]}
              iconClass="fa fa-trash-o delete-icon"
              onMouseEnter={[Function]}
              parentClass=""
              pointer={true}
            >
              <div
                className="icon-wrapper pointer"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-trash-o delete-icon"
                  data-target="#modal"
                  data-toggle="modal"
                >
                  <span />
                </i>
              </div>
            </Icon>
          </div>
        </div>
      </ListItem>
    </div>
  </List>
</RecipeList>
`;

exports[`Recipes list component  to match empty snapshot  1`] = `
<List
  className="recipe-list"
  style={Object {}}
>
  <h4
    className="display-4 bold header"
  >
    my recipes
  </h4>
  <ListItem
    className="recipe-list-item"
    content="recipe"
    handleClick={[Function]}
    index={0}
    key="1"
  >
    <div>
      <Icon
        active=""
        className=""
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        iconClass="fa fa-pencil edit-icon"
        onMouseEnter={[Function]}
        parentClass=""
        pointer={true}
      />
      <Icon
        active=""
        className=""
        dataTarget="#modal"
        dataToggle="modal"
        handleClick={[Function]}
        iconClass="fa fa-trash-o delete-icon"
        onMouseEnter={[Function]}
        parentClass=""
        pointer={true}
      />
    </div>
  </ListItem>
</List>
`;
